python/ray/util/sgd/v2/worker_group.py

```diff
@@ -22,6 +24,13 @@ def __execute(self, func: Callable[..., T], *args, **kwargs) -> T:
         return func(*args, **kwargs)
 
 
+@dataclass
+class Worker:
+    """Class representing a Worker."""
+    actor: ActorHandle
+    metadata: Dict
+
```

Comment:
Do we need this? Can we just call `get_node_id` in `Backend.start_training` method?

I'm also not opposed to creating this data class, but if we do then then we should be consistent and move _all_ data here (i.e. the methods currently defined in `utils.py`: `get_node_id`, `get_hostname`, `get_gpu_ids`). Then we can have clear phases of startup in which (1) all "worker data" is fetched upon instantiation, (2) the backend is setup, (3) worker data is used to populate the `Session`.

Also, if we go with this approach we should use another typed `dataclass` for the `metadata` so the API is clear!

Reply:
Good point, fixed!